components/infra-proxy-service/migrations/pipeline/utility.go

```diff
@@ -738,3 +743,60 @@ func createLocalUser(ctx context.Context, localUserClient local_user.UsersMgmtSe
 	}
 	return nil
 }
+
+// StoreOrgsUsersAssociation  populate the users association with orgs based on the actions
+func StoreOrgsUsersAssociation(ctx context.Context, st storage.Storage, result pipeline.Result) (pipeline.Result, error) {
+	log.Info("Starting with the storing org user association for migration id :", result.Meta.MigrationID)
+
+	var err error
+	selectedUsersMap := createMapForUsers(result.ParsedResult.Users)
+	for _, orgUsers := range result.ParsedResult.OrgsUsers {
+		for _, orgUserAssociation := range orgUsers.Users {
+			if _, keyPresent := selectedUsersMap[orgUserAssociation.Username]; keyPresent {
+				if orgUserAssociation.ActionOps == pipeline.Skip {
+					result.ParsedResult.OrgsUsersAssociationsCount.Skipped++
+					continue
+				}
+				err, _ = storeOrgUserAssociation(ctx, st, result.Meta.ServerID, orgUsers.OrgName.Name, orgUserAssociation)
+				if err != nil {
+					log.Errorf("Failed to migrate org user %s of org %s for migration id %s : %s", orgUserAssociation.Username, orgUsers.OrgName.Name, result.Meta.MigrationID, err.Error())
+					result.ParsedResult.OrgsUsersAssociationsCount.Failed++
+					continue
+				}
+				result.ParsedResult.OrgsUsersAssociationsCount.Succeeded++
+			}
+		}
+	}
+	if int64(len(result.ParsedResult.OrgsUsers)) == result.ParsedResult.OrgsUsersAssociationsCount.Failed {
+		return result, err
+	}
+	log.Info("Completed with the storing org user association for migration id :", result.Meta.MigrationID)
+	return result, nil
+}
+
+func createMapForUsers(users []pipeline.User) map[string]string {
+	usersMap := make(map[string]string)
+	for _, user := range users {
+		usersMap[user.Username] = ""
+	}
+	return usersMap
+}
+
+// storeOrgUserAssociation stores a single Org user into DB
+func storeOrgUserAssociation(ctx context.Context, st storage.Storage, serverID string, orgID string, orgUserAssociation pipeline.UserAssociation) (error, pipeline.ActionOps) {
+	var actionTaken pipeline.ActionOps
+	var err error
+	switch orgUserAssociation.ActionOps {
```

Comment:
```suggestion
	switch actionTaken = orgUserAssociation.ActionOps; orgUserAssociation.ActionOps {
```

This will remove unnecessary redefining actionTaken